src/client/service-worker/index.js

```diff
@@ -14,28 +14,28 @@
  * permissions and limitations under the License.
  */
 
-import { addErrorToReport } from '@americanexpress/one-app-ducks';
-
 export function importServiceWorkerClient(settings) {
   // In the future, we should consider making one-service worker added dynamically
   // as an external chunk with the ability to support conditional load.
-  return import(/* webpackChunkName: "pwa-client" */ './client')
+  return import(/* webpackChunkName: "service-worker-client" */ './client')
```

Comment:
Are we not using `async/await` because of bundle size or can we add that for more readability?

Reply:
The reason for not choosing to use `async/await` is that the nesting doesn't get deeper than one block (spaced out for readability) and using the syntax will add transforms that wrap `regeneratorRuntime` around the functions - although minuscule, it avoids adding extra wrappings and runtime overhead. I'd prefer using the native `Promise` implementation when possible on the client-side.

Reply:
I can understand the bundle size argument for this case, however, for most other cases `async/await` is much more readable, maintainable and worth the trade off for bundle size. As modern browsers have adopted `async/await` natively, legacy browsers will have a larger bundle size for polyfill but, most users will not incur a bundle hit.

Reply:
can we get away with not using `regeneratorRuntime` ? service worker is only supported in modern browsers, are they browsers that support service worker but not async await ?

I don't think that this is enough of an issue for me to block this PR

Reply:
Yes, this use case is fine here. We can move on.